File: opencv5.py
import cv2
import numpy as np

img = cv2.imread("1.jpg")

if img is None:#检查是否读取成功

    print("错误：无法读取图像")

    exit()

h, w = img.shape[:2]#获取图像尺寸

center = (w // 2, h // 2)#计算旋转中心点

M = cv2.getRotationMatrix2D(center, 45, 1.0)#计算旋转矩阵

x_img = cv2.warpAffine(img, M, (w, h))#旋转

cv2.imshow("TTT", x_img)

key = cv2.waitKey(0)#等待用户按键，参数0表示无限等待，直到用户按下任意键

if key == ord('s'):  #如果按下's'键

    cv2.imwrite('/home/abctfd/tu/6.jpg',x_img)#保存

    print("灰度图已保存: /home/abctfd/tu/6.jpg")

else:  #如果按下其他键

    print("图像未保存。")

File: opencv5.cpp
#include <opencv2/opencv.hpp>
#include <iostream>
using namespace cv;
using namespace std;

int main() {

    Mat img = imread("1.jpg");//读取图像

    if (img.empty()) {//检查是否读取成功

        cout << "错误：无法读取图像" << endl;

        return -1;

    }
    
    int h = img.rows;//获取图像尺寸

    int w = img.cols;//获取图像尺寸
    
    Point2f center(w/2.0f,h/2.0f);//计算旋转中心点
    
    Mat M = getRotationMatrix2D(center,45,1.0);//计算45度旋转的变换矩阵
    
    Mat X_img;

    warpAffine(img,X_img,M,img.size());//旋转
    
    namedWindow("TTT",WINDOW_NORMAL);

    imshow("TTT",X_img);
    
    while(true){//按q键关闭窗口

        int key = waitKey(0);

        if(key == 'q'){

            break;

        }
    }

    destroyAllWindows();

    return 0;
}